haskell-customize.el

```diff
@@ -66,9 +67,8 @@ Otherwise if there's a *.cabal file, cabal-repl will be used.
 If none of the above apply, ghci will be used."
   :type '(choice (const auto)
                  (const ghci)
-                 (const cabal-repl)
                  (const stack-ghci)
-                 (const cabal-new-repl))
```

Comment:
Seems reasonable to keep the value _'cabal-new-repl_ and emit _"cabal new-xxx"_
What about supporting _"cabal v1-xxx"_ and _"cabal v2-xxx"?_  My preference is don't, as it makes the elisp code too complicated, and it is so easy for the user to edit the generated command before it is run.
In general, is it worth detecting and coping with the user's version of _cabal_?

Reply:
Not worth detecting, and I'm fine with only one version of "cabal repl" being supported, ie. I don't think we need to make it switchable between "repl" and "new-repl". However, what I'm wary of is that somebody has configured a setting that then silently has no effect. I would rather they get a hard error, or have the `cabal-new-repl` value treated as `cabal-repl`, perhaps with a warning message.

Reply:
OK. I will restore  _'cabal-new_ and _'cabal-new-repl_, which will emit _cabal new-build ..._ and _cabal new-repl ..._,
as _cabal_ still supports them. Perhaps they'll be needed again when v3 is introduced.

Reply:
Did that, lbut it ooks too complicated. After reading your comments again,  I agree with not emitting _cabal new-repl_ or _cabal new-build_, and just accepting the obsolete customization enum 'cabal-new-repl with a warning.
I will clean up my branch -- in fact, collapse it into one commit. I hope that's ok, this is my 1st pull request.

Reply:
> I will clean up my branch -- in fact, collapse it into one commit. I hope that's ok, this is my 1st pull request.

It's not super important for it to be a single commit, tbh, because I can easily squash it when merging if necessary. But a bit of squashing on your end just to keep the commits clean is of course fine, and helps preserve the history.